ADR/ADRYYY-store-allowed-domain-list-as-file-in-forms-deploy.md

```diff
@@ -0,0 +1,126 @@
+
+However, there are two downsides to this approach: One, this is configuration data that is defines how the service is run, and should be stored in a way which allows control, auditing and replication independent of the runtime data. Two, these lists need to be used in two different places, in the identity provider that authenticates users email addresses as well as the `forms-admin` app.
+
+```mermaid
+%%{init: {"flowchart": {"defaultRenderer": "elk"}} }%%
+flowchart TD
+    subgraph forms-admin
+    start([User visits GOV.UK Forms])
+    --> Auth0
+
+    subgraph Auth0
+    login_id_prompt[Ask user for email address]
+    -->
+
+    restrict_users_to_allowed_domains{Is the email address domain<br>in the allow list?}
+    -- No -->
+    unauthorised_email_domain([Show error<br>'Enter a government email address'])
+
+    restrict_users_to_allowed_domains -- Yes -->
+    authenticate_email_address[[Check that the user has<br>access to that email inbox]]
+    end
+
+    authenticate_email_address -->
+    find_or_create_user{Has the user signed in before?}
+
+    find_or_create_user -- No -->
+    create_user[Create a new user record]
+    -->
+    deny_users_with_restricted_domains{Is the email address domain<br>in the restricted list?}
+
+    deny_users_with_restricted_domains -- No -->
+    set_has_access_true[Allow access for the new user]
+    --> has_access
+
+    deny_users_with_restricted_domains -- Yes -->
+    set_has_access_false[Deny access for the new user]
+    --> has_access
+
+    find_or_create_user -- Yes -->
+    has_access{Does the user have access?}
+
+    has_access -- No -->
+    access_denied([Show error 'You do not have permission'])
+    has_access -- Yes -->
+    govuk_forms_home_page([Welcome to GOV.UK Forms])
+    end
+
+```
+
+The identity provider we are using currently is Auth0; it is configured using Terraform code in the `forms-deploy` repository. The `forms-admin` app is built and run as a container. The container image is built from code in the `forms-admin` repository and deployed as part of a release pipeline. This pipeline is defined using code in the `forms-deploy` repo, and also itself retrieves the `forms-deploy` repo to determine how the `forms-admin` app should be deployed. Getting data from the databases into Auth0 is more difficult compared to getting data into the app during deployments. This suggests a solution where a single source-of-truth ouside of the database is access by code in the `forms-deploy` repo both when Auth0 is configured by terraform and when the deployment pipeline is run.
+
+## Decision
+
+> what the team has decided to do
+
+Add configuration file(s) (YAML or JSON) to forms-deploy repo with list of domains and associated organisations
+    - Suggest folder `forms-deploy/config/data`
+
+Add Rake task to load domain configuration into database from file or environment variable
```

Comment:
Initially I thought storing the email domains in the database would be overcomplicating this versus just using parameter store, at least until we have a concrete need for something more sophisticated.

However, that need for something more sophisticated is rapidly approaching if we are going to be automating some part of the organisation assignment based on email domain. If we need that functionality, I think what is proposed here is going to work well.